old_code.py

```diff
@@ -0,0 +1,107 @@
+from PyQt4.QtCore import *
+from qgis.networkanalysis import *
+from qgis.core import *
+import processing
+from processing.tools.vector import VectorWriter
+import psycopg2
+
+
+point_layer = processing.getObject("/home/anshulgoel031gmailcom/Downloads/sudanshu/projections.shp")
+network_layer = processing.getObject("/home/anshulgoel031gmailcom/Downloads/bangaloreStreets.shp")
+
+#creating an empty layer
+
+cLayer = qgis.utils.iface.mapCanvas().currentLayer()
+provider = cLayer.dataProvider()
+print provider.encoding(),",", provider.fields(),",", provider.crs()
+writer = QgsVectorFileWriter( "/home/anshulgoel031gmailcom/Downloads/MTP1v2/sudanshu.shp", provider.encoding(), provider.fields(),QGis.WKBLineString, provider.crs() )
+#writer = VectorWriter("/home/anshulgoel031gmailcom/Downloads/MTP1v2/rs_generic_10062019_3.shp", None, [QgsField("order", QVariant.Int)], network_layer.dataProvider().geometryType(), network_layer.crs() )
+
+layer = QgsVectorLayer("/home/anshulgoel031gmailcom/Downloads/MTP1v2/sudanshu.shp","rs10062019", "ogr")
+
+if not layer.isValid():
+    print "Layer failed to load!"
+#start editing
+layer.startEditing()
+provider = layer.dataProvider()
+#add fields
+provider.addAttributes( [ QgsField("AddressFrom",  QVariant.String)] )
+provider.addAttributes( [ QgsField("AddressTo",  QVariant.String)] )
+layer.commitChanges()
+
+# prepare graph
+vl = network_layer
+director = QgsLineVectorLayerDirector( vl, -1, '', '', '', 3 )
+properter = QgsDistanceArcProperter()
+director.addProperter( properter )
+crs = vl.crs()
+builder = QgsGraphBuilder( crs )
+
+#
+
+#builder = QgsGraphBuilder(layer.crs()) #sudhanshu one
+#
+#changes may have to be done in builder currently taken of sudhanshu
+
+# prepare points
+features = processing.features(point_layer)
+point_count = point_layer.featureCount() #total no. of records in my case
+
+points = []
+
+for f in features:
+  points.append(f.geometry().asPoint())
+
+tiedPoints = director.makeGraph( builder, points )
+graph = builder.graph()
+
+for i in range(0,point_count-1):
+    for j in range (i+1,point_count ):
+            
+    #for i in range(0,5):    
+    #    progress.setPercentage(int(100 * i/ point_count))
+        
+        from_point = points[i]
+        to_point = points[j]
+    
+        from_id = graph.findVertex(from_point)
+        to_id = graph.findVertex(to_point)
+    
+        (tree,cost) = QgsGraphAnalyzer.dijkstra(graph,from_id,0)
+    
+        if tree[to_id] == -1:
+            continue # ignore this point pair
+        else:
+            # collect all the vertices between the points
+            route_points = []
+            curPos = to_id 
+            while (curPos != from_id):
+                route_points.append( graph.vertex( graph.arc( tree[ curPos ] ).inVertex() ).point() )

```

Comment:
What can be used instead of `graph.arc`? 

I tried to use `graph.incomingEdges ` but it is now working.

Reply:
How about `graph.edge(tree[curPos]).fromVertex().point()`?

The `QgsGraph.arc()` method is renamed to `QgsGraph.edge()` in version 3x.  And the `QgsGraphEdge.inVertex()` is renamed to `QgsGraphEdge.fromVertex()`.  See https://api.qgis.org/api/api_break.html for more such renames.